clang/test/CIR/CodeGen/array-init.c

```diff
@@ -8,3 +8,22 @@ void foo() {
 // CHECK-NEXT: %1 = cir.const(#cir.const_array<[9.000000e+00, 8.000000e+00, 7.000000e+00]> : !cir.array<f64 x 3>) : !cir.array<f64 x 3>
 // CHECK-NEXT: cir.store %1, %0 : !cir.array<f64 x 3>, cir.ptr <!cir.array<f64 x 3>>
 
+void bar(int a, int b, int c) {
+  int arr[] = {a,b,c};
+}
+
+// CHECK: cir.func @bar
+// CHECK:      [[ARR:%.*]] = cir.alloca !cir.array<!s32i x 3>, cir.ptr <!cir.array<!s32i x 3>>, ["arr", init] {alignment = 4 : i64}
```

Comment:
Constant arrays are pretty common thing that gets used, I think it would be useful to raise it into a `cir.init_array(op)` and hide much of this complexity, how about you land this and do a follow up PR implementing such new op (which is then lowers to this current output out of `LoweringPrepare`)?

(cc @sitio-couto)

Reply:
It seems logical! I can implement this new opcode.But I'm new in clang IR and

Reply:
No worries, we can help there!